README.md

```diff
@@ -80,6 +80,32 @@ let documents = vec![
 
  println!("Embeddings length: {}", embeddings.len()); // -> Embeddings length: 4
  println!("Embedding dimension: {}", embeddings[0].len()); // -> Embedding dimension: 384
+
+// reranker model usage 
+
+// simple compute reranker scores
+let result = TextRerank::try_new(RerankerModel::BGERerankerBase,
+InitOptions { show_download_progress: true, ..Default::default() }).unwrap();
+
+let texts = vec![
+    ("what is panda?", "hi"),
+    ("what is panda?", "The giant panda (Ailuropoda melanoleuca), sometimes called a panda bear or simply panda, is a bear species endemic to China."),
+    ("what is panda?", "panda is animal"),
+    ("what is panda?", "i dont know"),
+    ("what is panda?", "kind of mammal"),
```

Comment:
I am unsure of such an interface?

Shouldn't it be a single query and list of documents? Instead of a list of query and document tuples?

Assuming we're reranking a query and its candidates?

Reply:
Yes, I was considering the same question. 
However, according to the sample code on the original model page, they also use tuples. 
While it's possible to use one query and a list of documents, I'm concerned there might be other usage scenarios. Therefore, I've decided to follow the original sample code.
`from FlagEmbedding import FlagReranker
reranker = FlagReranker('BAAI/bge-reranker-large', use_fp16=True) # Setting use_fp16 to True speeds up computation with a slight performance degradation

score = reranker.compute_score(['query', 'passage'])
print(score)

scores = reranker.compute_score([['what is panda?', 'hi'], ['what is panda?', 'The giant panda (Ailuropoda melanoleuca), sometimes called a panda bear or simply panda, is a bear species endemic to China.']])
print(scores)`

Reply:
Cohere's interface(Only one I've used) seems idiomatic.

https://docs.cohere.com/docs/reranking

We get the sorted scores with doc text and index.

Reply:
I just checked the coherence implementation, it's great. Should we change the rerank method just like that?

Reply:
I +1 that.